tests/connectionpool_test.go

```diff
@@ -137,3 +143,166 @@ func TestConnectionPool(t *testing.T) {
 		return err
 	}))
 }
+
+// TestConnectionPoolWithReuseInboundUser verifies that a ConnectionPool can be created without
+// specifying a username, which enables the "Reuse Inbound User" feature. This allows the
+// connection pool to use the credentials of whoever is connecting to the pool rather than
+// a fixed service user.
+func TestConnectionPoolWithReuseInboundUser(t *testing.T) {
+	t.Parallel()
+	defer recoverPanic(t)
+
+	var (
+		ctx, cancel = context.WithTimeout(context.Background(), 10*time.Minute)
+		pgName      = randName("pg")
+		dbName      = randName("database")
+		poolName    = randName("connection-pool-inbound")
+
+		s = NewSession(ctx, k8sClient, cfg.Project)
+
+		findPoolFunc = func(projectName, serviceName, poolName string) *service.ConnectionPoolOut {
+			var avnPool *service.ConnectionPoolOut
+			services, err := avnGen.ServiceGet(ctx, cfg.Project, pgName)
+			require.NoError(t, err)
+
+			for _, p := range services.ConnectionPools {
+				if p.PoolName == poolName {
+					avnPool = &p
+					break
+				}
+			}
+
+			return avnPool
+		}
+	)
+
+	defer func() {
+		cancel()
+		s.Destroy(t)
+	}()
```

Comment:
For some reason this test doesn't cleanup the PG resource. Could you double check this?

Reply:
Fixed